File: resources/scripts/battle-formulas.lua
specialized = {}

function specialized:getEffectiveStat(pokemon, stat, baseValue)
    if stat == stats.speed and external.hasFlag(pokemon.id, "Paralysis") then
        baseValue = baseValue * 0.5
    end

    return baseValue
end

--[[
    Given the index of a move that has enough PP, checks if it can be used. Moves
    like Disable, items like Choice Band, etc should be considered.
    Signature: bool(Pokemon, int)
]]
function specialized:canUseMove(pokemon, moveIndex)
    -- TODO
    return true
end

File: resources/scripts/original/battle-formulas.ts
import { Stat } from './enums';
import { external, Pokemon } from './types';

export namespace specialized {
    export function getEffectiveStat(pokemon: Pokemon, stat: Stat, baseValue: number): number {
        if (stat == Stat.Speed && external.hasFlag(pokemon.id, 'Paralysis')) {
            baseValue *= 0.5;
        }

        return baseValue;
    }

    /**
     * Given the index of a move that has enough PP, checks if it can be used. Moves
     * like Disable, items like Choice Band, etc should be considered.
     */
    export function canUseMove(pokemon: Pokemon, moveIndex: number): boolean {
        // TODO
        return true;
    }
}
